tests/core/tau_test.py

```diff
@@ -0,0 +1,169 @@
+import unittest
```

Comment:
numpy isn't mocked and the doc comments are missing, i will review the file further after this is fixed